python/lsst/pipe/tasks/matchBackgrounds.py

```diff
@@ -19,683 +19,627 @@
+    numSigmaClip = Field[int](
+        doc="Sigma for outlier rejection. Ignored if ``gridStatistic != 'MEANCLIP'``.",
+        default=3,
     )
-    bestRefWeightCoverage = pexConfig.RangeField(
-        dtype=float,
-        doc="Weight given to coverage (number of pixels that overlap with patch), "
-        "when calculating best reference exposure. Higher weight prefers exposures with high coverage."
-        "Ignored when reference visit is supplied",
-        default=0.4,
-        min=0., max=1.
-    )
-    bestRefWeightVariance = pexConfig.RangeField(
-        dtype=float,
-        doc="Weight given to image variance when calculating best reference exposure. "
-        "Higher weight prefers exposures with low image variance. Ignored when reference visit is supplied",
-        default=0.4,
-        min=0., max=1.
+    numIter = Field[int](
+        doc="Number of iterations of outlier rejection. Ignored if ``gridStatistic != 'MEANCLIP'``.",
+        default=3,
     )
-    bestRefWeightLevel = pexConfig.RangeField(
-        dtype=float,
-        doc="Weight given to mean background level when calculating best reference exposure. "
-        "Higher weight prefers exposures with low mean background level. "
-        "Ignored when reference visit is supplied.",
-        default=0.2,
-        min=0., max=1.
-    )
-    approxWeighting = pexConfig.Field(
-        dtype=bool,
-        doc=("Use inverse-variance weighting when approximating background offset model? "
-             "This will fail when the background offset is constant "
-             "(this is usually only the case in testing with artificial images)."
-             "(usePolynomial=True)"),
+
+    approxWeighting = Field[bool](
+        doc="Use inverse-variance weighting when approximating the background offset model? This will fail "
+        "when the background offset is constant (usually only the case in testing with artificial images)."
+        "Only applied if ``usePolynomial=True``.",
         default=True,
     )
-    gridStdevEpsilon = pexConfig.RangeField(
+    gridStdevEpsilon = RangeField(
         dtype=float,
-        doc="Tolerance on almost zero standard deviation in a background-offset grid bin. "
-        "If all bins have a standard deviation below this value, the background offset model "
-        "is approximated without inverse-variance weighting. (usePolynomial=True)",
+        doc="Tolerance on almost zero standard deviation in a background-offset grid bin. If all bins have a "
+        "standard deviation below this value, the background offset model is approximated without "
+        "inverse-variance weighting. Only applied if ``usePolynomial=True``.",
         default=1e-8,
-        min=0.
+        min=0.0,
     )
 
 
-class MatchBackgroundsTask(pipeBase.Task):
+class MatchBackgroundsTask(PipelineTask):
+    """Match the backgrounds of a list of warped exposures to a reference.
+
+    This task is a part of the background subtraction pipeline.
```

Comment:
Put this paragraph in a `Notes` section; the text right below the quick summary is supposed to just be a slightly longer summary (for reasons I don't understand, but that's how it is).